cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts

```diff
@@ -0,0 +1,266 @@
+  }
+  // If none match fallback to string;
+  return typedValue;
+}
+
+/**
+ * Determines type for the value of a property based on the type
+ * provided by the backend.
+ */
+function getValueFromBackendType(type: string, propertyValue: string): IPropertyValueType {
+  if (!type) {
+    return inferTypeFromValue(propertyValue);
+  }
+  switch (type) {
+    case 'boolean':
+      return propertyValue === 'true' ? true : false;
+    case 'int':
+    case 'long':
+      return isNaN(Number(propertyValue)) ? propertyValue : Number(propertyValue);
+    case 'string':
+    default:
+      // We still infer the type here as a 'string' type from backend
+      // sometime may refer to boolean.
+      return inferTypeFromValue(propertyValue);
+  }
+}
+
+/**
+ * Converts <string, string> map to <string, int|long|boolean|string> map
+ * This is necessary to pass in the right context to jexl to validate
+ * the expression.
+ *
+ * For example,
+ * jexl expression: 'property1 == true'
+ * context: { property1: 'true', property2: 'string2' }
+ * If the context is passed as is to the jexl library the above expression
+ * will be false as 'true' !== true. So we need to make a best guess
+ * in converting the string to appropriate types for the jexl expression
+ * to be parsed correctly.
+ *
+ * @returns typedValues - properties map with values being appropriate type(Boolean, number or string)
+ */
+function getTypedPropertyValues(
+  propertyValues: IPropertyValues,
+  propertiesFromBackend: PluginProperties
+): IPropertyTypedValues {
+  const typedValues = {};
+  if (propertyValues && typeof propertyValues === 'object') {
+    Object.keys(propertyValues).forEach((property) => {
+      const propertyConfigFromBackend = propertiesFromBackend[property];
+      if (!propertyConfigFromBackend) {
+        return;
+      }
+      const type = propertyConfigFromBackend.type;
+      const value = propertyValues[property];
+      if (isValueMacro(value)) {
+        return value;
+      }
+      typedValues[property] = getValueFromBackendType(type, value);
+    });
+  }
+  return typedValues;
+}
+
+/**
+ * Filter properties based on filtering condition. Will hide all the
+ * properties by default that are part of each filter. If the condition is true
+ * then show all the properties as part of the filter's show.
+ *
+ * Will skip all required properties from being hidden.
+ *
+ * If the property involved in the condition is a macro skip chekcing for condition
+ */
+export function filterByCondition(
+  filteredGroupConfiguration: IConfigurationGroup[],
+  widgetJSON: IWidgetJson,
+  propertiesFromBackend: PluginProperties,
+  propertyValues: IPropertyValues
+): IFilteredConfigurationGroup[] {
+  // By default every property will be shown
+  let propertiesToShow: string[] = flatten(
+    filteredGroupConfiguration.map((group) => {
+      return group.properties.map((prop) => prop.name);
+    })
+  );
+  const filters: IPropertyFilter[] = widgetJSON.filters;
```

Comment:
should handle when widgetJSON is null/undefined

Reply:
Just an FYI this won't be executed when there is no widgetJSON right? We would render everything as textbox and not bother about doing anything this route. If I understand correctly if there is no widgetJSON we just iterate through all the plugin properties and render textboxes.

Anyhow have added a check just to be defensive.

Reply:
I think it will still be called, there is no prevention from the ConfigurationGroup useEffect hook. The only condition where this would not get called is when the pluginProperties is undefined. But since the call is being wrapped in a try/catch block, it would've been fine

Reply:
Talked to @elfenheart . The current angular implementation wouldn't call this when there is no widgetJSON. However we haven't decided on how things will work when we refactor the pipelines to be in React. We can decide on how the control flows when we refactor. Have added the safety check for now.